### PR TITLE
client: cleanup formatting button component

### PR DESCRIPTION
**Pull request by lordvins226**
#9 for cleanup formatting button , i create a hook .

### CODE REVIEW

File: client/components/document/hooks/use-formatting-click.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import type { RefObject } from "react";
```
Comment (MaxLeiter) [inline client/components/document/hooks/use-formatting-click.ts:1]: This is great but can we keep it in lib/?

Comment (lordvins226) [reply client/components/document/hooks/use-formatting-click.ts:1]: > This is great but can we keep it in lib/?

Off course

**Comment by MaxLeiter** [comment]
It seems like the ability to click the buttons without highlighting text has disappeared. We should insert `**<cursor>**` if `bold` is selected, for example. tested on Firefox + macOS

**Comment by lordvins226** [comment]
> It seems like the ability to click the buttons without highlighting text has disappeared. We should insert **<cursor>** if bold is selected, for example. tested on Firefox + macOS. Thanks for your work!

I will take a look 👍🏽

**Comment by lordvins226** [comment]
@MaxLeiter can we open an issue for cursor feature ? I fixed the ability to click without highlighting text

**Comment by MaxLeiter** [comment]
I’d rather solve it here if that’s fine with you. Whats the issue youre experiencing?

**Comment by lordvins226** [comment]
Hello @MaxLeiter . I tried several solutions to fix the cursor, but none work. Do you have an approach to fix the problem ?

**Comment by MaxLeiter** [comment]
Thanks a lot for your work @lordvins226 but I found https://github.com/Resetand/textarea-markdown-editor and have decided to use that. It looks like the amount of code to keep this in-house isn't very feasible, and that library has solid tests and a reasonable code-base. If you want to see where they accomplished what we were working on here, take a look at https://github.com/Resetand/textarea-markdown-editor/blob/master/src/lib/Cursor.ts

**Comment by lordvins226** [comment]
Hey @MaxLeiter. This library is Interesting 🤔. is it already integrated?

**Pull request closed without merging**